in_toto/user_settings.py

```diff
@@ -0,0 +1,132 @@
+"""
+<Program Name>
+  user_settings.py
+
+<Author>
+  Lukas Puehringer <lukas.puehringer@nyu.edu>
+
+<Started>
+  Oct 25, 2017
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  Provides methods to read environment variables and rcfiles and to write
+  them to the in_toto.settings module (override them).
+
+  Notes:
+  - Variable values are converted to lists if they contain colons
+  - Variable names are converted to upper case
+  - Environment variables need to be prefixed with "IN_TOTO_". The prefix is
+    removed when the variable is written to the settings module,
+    e.g. the envvar...
+    IN_TOTO_ARTIFACT_EXCLUDES="*.link:.gitignore"
+
+    becomes the setting ...
+    in_toto.settings.ARTIFACT_EXCLUDES = ["*.link", ".gitignore"]
+
+  - RCfile values follow the format expected by Python's builtin
+    ConfigParser. The format requires at least one section, sections however
+    are ignored, e.g. the rcfile...
+    [ignored section name]
+    artifact_basepath = "/a/b/c"
+
+    becomes the setting ...
+    in_toto.settings.ARTIFACT_BASEPATH = "/a/b/c"
+
+  - Order of precedence
+    1. RCfiles
+        .in_totorc
+        ~/.in_totorc
+        ~/.in_toto/config
+        ~/.config/in_toto
+        ~/.config/in_toto/config
+        /etc/in_totorc
+        /etc/in_toto/config
+    2. Environment Variables
+    3. Settings defined in `settings.py`
+
+"""
+import os
+import ConfigParser
+import in_toto.settings as SETTINGS_MODULE
+
+ENV_PREFIX = "IN_TOTO_"
+USER = os.path.expanduser("~")
+
+RC_PATHS = [
+  os.path.join("/etc", "in_toto", "config"),
```

Comment:
Can you list these in the order of precedence used in the docstring?  It helps readability, and a reviewer to verify that an RC path isn't missing.

Reply:
I guess I rather reverse the list in the docstring. If I reversed here, I'd have to do a `RC_PATHS.reverse()` below where I use the list, or iterate backwards, which seems odd too.